complete/src/main/java/hello/PubsubApplication.java

```diff
@@ -0,0 +1,89 @@
+package hello;
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.gcp.pubsub.core.PubsubTemplate;
+import org.springframework.cloud.gcp.pubsub.support.GcpHeaders;
+import org.springframework.cloud.gcp.pubsub.support.SubscriberFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.annotation.MessagingGateway;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.gcp.AckMode;
+import org.springframework.integration.gcp.inbound.PubsubInboundChannelAdapter;
+import org.springframework.integration.gcp.outbound.PubsubMessageHandler;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+
+@SpringBootApplication
+public class PubsubApplication {
```

Comment:
The user can surely do it that way, and it will still work. But as best practices go, I think this is how we best separate concerns -- configuration on one side, controller on the other.